pkg/storage/engine/disk_map_test.go

```diff
@@ -220,6 +220,84 @@ func TestRocksDBMapSandbox(t *testing.T) {
 	})
 }
 
+// TestRocksDBStore tests that the allowDuplicates setting allows duplicate
+// keys to be put.
+func TestRocksDBStore(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+	tempEngine, err := NewTempEngine(ctx, base.DefaultTestStoreSpec)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tempEngine.Close()
+
+	var (
+		v1 = []byte("v1")
+		v2 = []byte("v2")
+		k1 = []byte("k1")
+	)
+
+	tests := []struct {
+		fn func(e Engine) *RocksDBMap
+		// expect is a map containing the expected number of found values for key k1.
+		expect map[string]int
+	}{
+		{
+			NewRocksDBStore,
+			map[string]int{
+				string(v1): 4,
+				string(v2): 2,
+			},
+		},
+		{
+			NewRocksDBMap,
+			map[string]int{
+				string(v1): 1,
+				// v1 is the final Put, so it should overwrite the previous v2.
+				string(v2): 0,
+			},
+		},
+	}
+
+	for ti, tc := range tests {
+		t.Run(fmt.Sprint(ti), func(t *testing.T) {
```

Comment:
It would be nice to have a more descriptive test name. Maybe add this to the test case struct (i.e. "AllowDuplicates=true").